src/Package.php

```diff
@@ -129,17 +147,25 @@ private function deduceTrackingCode(): Package
         $tracking_code = $this->tracking_code;
         $carrier_code  = null;
         $provider_code = null;
+        $web_link = null;
 
         if (preg_match('/^[0-9]{2}[0-9]{4}[0-9]{4}$/', $tracking_code, $matches)) {
             $carrier_code = Carrier::CODE_DHL;
-        } elseif (preg_match('/^[1-9]{4}[0-9]{4}[0-9]{4}$/', $tracking_code, $matches)) {
+            $web_link = "https://www.dhl.com/en/express/tracking.html?AWB=".$matches[0];
+        } elseif (preg_match('/^[0-9]{12}$|^[0-9]{15}$/', $tracking_code, $matches)) {
             $carrier_code = Carrier::CODE_FEDEX;
-        } elseif (preg_match('/^1Z[A-Z0-9]{3}[A-Z0-9]{3}[0-9]{2}[0-9]{4}[0-9]{4}$/i', $tracking_code)) {
+            $web_link = "https://www.fedex.com/fedextrack/?tracknumbers=" . $matches[0] .
+                         "&cntry_code=us&language=english";
```

Comment:
@bell2041 We might be able to simplify the concatenation of these URLs. Can you test these to see if we can omit some of the parameters like `&cntry_code=us&language=english`? That would make the URL much shorter and I bet it'll still work - but we should test for each carrier.